modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -1154,15 +1154,11 @@ private CPDefinition _updateNestedResources(
 			product.getProductSpecifications();
 
 		if (productSpecifications != null) {
-			_cpDefinitionSpecificationOptionValueService.
-				deleteCPDefinitionSpecificationOptionValues(
-					cpDefinition.getCPDefinitionId());
-
 			for (ProductSpecification productSpecification :
 					productSpecifications) {
 
 				ProductSpecificationUtil.
-					addCPDefinitionSpecificationOptionValue(
+					addOrUpdateCPDefinitionSpecificationOptionValue(
```

Comment:
here reindex are as many as the spec values passed.
If after this upsert I need to also do the deletion there may be more reindex compared to the previous implementation